### ISSUE TITLE
how to access the tensorpack layers and pass to tensorflow for more processing 

### ISSUE THREAD

**Issue by prashantsingh905** [ppwwyyxx/tensorpack#204]
I have the following model using resnet-imagenet. 
   logits = (LinearWrap(self.x)
                      .Conv2D('conv0', 64, 7, stride=2, nl=BNReLU)
                      .MaxPooling('pool0', shape=3, stride=2, padding='SAME')
                      .apply(layer, 'group0', block_func, 64, defs[0], 1, first=True)
                      .apply(layer, 'group1', block_func, 128, defs[1], 2)
                      .apply(layer, 'group2', block_func, 256, defs[2], 2)
                      .apply(layer, 'group3', block_func, 512, defs[3], 2)
                      .BNReLU('bnlast')
                      .GlobalAvgPooling('gap')
                      .FullyConnected('fc0', 512, nl=tf.nn.relu)
                      .FullyConnected('linear', 1000, nl=tf.identity)())

now i want to access the fc0 layer and pass their content to my other function which is in tensorflow without tensorpack

  num_inputs= int(net.logits.fc0.output.get_shape()[1])
`w, b = net.logit.get_fc_weights(layer_index=16, net_data=None,
                                          num_inputs=num_inputs,
                                          num_outputs=n_joints * 2,
                                          wights_std=0.01,
                                          bias_init_value=0.0)` 
                net.fc_regression = tf.nn.xw_plus_b(net.fc7_dropout, w, b, name=scop

**Comment by ppwwyyxx** [comment]
LinearWrap is a syntax sugar useful when you don't need intermediate layer output.
If you need fc0 layer output, just stop LinearWrap at fc0.

**Comment by prashantsingh905** [comment]
Thanks for your reply. On the same note can you tell me how do i access the weights and biases from the previous layers define in linear warp. I tried to find it in docs but didn't get any info on this.

**Comment by ppwwyyxx** [comment]
There isn't a very straightforward way because I don't know where it can be added. 
Currently the docs tell you the name of the variables, which are 'W', 'b': http://tensorpack.readthedocs.io/en/latest/modules/models.html#tensorpack.models.Conv2D

Then you can access the tensor by, e.g. `t = Graph.get_tensor_by_name('conv0/W:0')`, assuming the name of the layer is 'conv0'.

**Comment by PatWie** [comment]
What about patching the layers:
```python
@layer_register()
def Conv2D(x, out_channel, kernel_shape,
           padding='SAME', stride=1,
           W_init=None, b_init=None,
           nl=tf.identity, split=1, use_bias=True,
           data_format='NHWC'):
# ...
return_value = nl(tf.nn.bias_add(conv, b, data_format=data_format) if use_bias else conv, name='output')
return_value.W = W
return_value.b = b
return return_value
```

Then one can write

```python
l = Conv2D('conv1', image, out_channel=96, kernel_shape=11, stride=4, padding='VALID')
print sess.run(l.W)
```

Not sure, how this behaves in multi-tower settings.

**Comment by ppwwyyxx** [comment]
@PatWie Yes that's my first thought as well and the interface would be very simple. But since tf.Tensor doesn't really have such attributes this may not be a safe option.

**Comment by ppwwyyxx** [comment]
Closing this and continue as a feature request in #228 .
